apigw.go

```diff
@@ -0,0 +1,43 @@
+package algnhsa
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"path"
+
+	"github.com/aws/aws-lambda-go/events"
+)
+
+var (
+	errNonAPIGateway = errors.New("non APIGatewayProxyRequest event")
+)
+
+func newAPIGatewayRequest(ctx context.Context, payload []byte, opts *Options) (lambdaRequest, error) {
+	var event events.APIGatewayProxyRequest
+	if err := json.Unmarshal(payload, &event); err != nil {
+		return lambdaRequest{}, err
+	}
+	if event.RequestContext.AccountID == "" {
```

Comment:
Is this all we have to key off of to determine if this event is from an APIGateway?